### ISSUE TITLE
watchLocation and getCurrentPosition usage conflicts

### ISSUE THREAD

**Issue by OlivierVoyer** [Agontuk/react-native-geolocation-service#172]
Hi, I am using both features of your package, watchLocation to detect when I am going to far from a reference position, in background, with low battery consumption, and getCurrentPosition from time to time to get an update of my current location (either in foreground or in background).

My issue is that it seems that the options I am using for the watch are overriden by my getPosition call (especially distanceFilter for example) and so my watch becomes crazy after the first getCurrentPositionCall as it does not take a significant filter after.

Is there a way to use both calls without the options being shared?

Thanks for your help.

Watch setup (once for all):

```
const watchId = Geolocation.watchPosition(
      (position) => {
        console.log('Geolocation new position:', position);
      },
      (error) => {
        console.log('Geolocation error:', error.message);
      },
      {
        timeout: 15000,
        maximumAge: 10000,
        enableHighAccuracy: true,
        distanceFilter: 50,
        interval: 60000,
        forceRequestLocation: true
      },
    );
```

Get Current Position (sometimes in either foreground or background):

```
Geolocation.getCurrentPosition(
          position => {
            console.log('position', position);
          },
          error => {
            console.warn('Location error:', error.message);
          },
          {
            timeout: 15000,
            maximumAge: 0,
            enableHighAccuracy: true,
            distanceFilter: 0,
            forceRequestLocation: true,
          },
        );
```

**Comment by oakis** [comment]
Why not use only `watchPosition` for both? They should both give the same position anyways..

**Comment by OlivierVoyer** [comment]
My situation is having a watch running all the way during my app execution and in background. And moving too far, so some other events, triggering a report with the need of the live position (getCurrentPosition). That is why I used the 2 different calls.

**Comment by oakis** [comment]
I mean use the same watchPosition for both. Store it globally if needed, for example with redux.
Then you can access it for both your needs, as the position should be the same.

**Comment by OlivierVoyer** [comment]
In my particular situation it cannot work.
My watch is running constantly to detect that the user moves more from xxx meters (so its distance filter is somewhat xxx/2).
But I also need to trigger some actions every xxx minutes that requires the precise current location (so its distance filter being 0 to avoid getting a timeout cause the user didn't move enough).
So when my getCurrentPosition is called (can be Headless or through a background task), the distanceFilter of my watch becomes 0 and start triggering on every single movement.

**Comment by Agontuk** [comment]
Not being able to reproduce this. I've tested in the emulator with the config you provided. `watchPosition` is only firing first time, until I change the location it's not emitting new position. Then I  called `getCurrentPosition` and it returned a position properly. But `watchPosition` still does not firing any events, which is ok.

Can you check again if it's something on your end ?

**Comment by OlivierVoyer** [comment]
Better to give an example probably, please find a sample project in attachment. Run `yarn install && cd iOS && pod install && cd ..` before running it.

[MyApp.zip](https://github.com/Agontuk/react-native-geolocation-service/files/4496801/MyApp.zip)

Note that I am running in on an iOS Simulator iOS 13 with location debug to "run" for example and get the following execution trace showing that the watch is corrupted as soon as I press the get location button:

```
[Sat Apr 18 2020 16:45:33.678]  LOG      Running "MyApp" with {"rootTag":11,"initialProps":{}}
[Sat Apr 18 2020 16:45:33.678]  LOG      Starting watch
[Sat Apr 18 2020 16:45:33.700]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 103.73, "latitude": 37.33002981, "longitude": -122.0208385, "speed": 3.6}, "timestamp": 1587221133256.74}
[Sat Apr 18 2020 16:45:41.935]  LOG      Fetching current location
[Sat Apr 18 2020 16:45:42.341]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 110.11, "latitude": 37.3299178, "longitude": -122.02044999, "speed": 3.46}, "timestamp": 1587221142338.309}
[Sat Apr 18 2020 16:45:42.342]  LOG      position {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 110.11, "latitude": 37.3299178, "longitude": -122.02044999, "speed": 3.46}, "timestamp": 1587221142338.309}
[Sat Apr 18 2020 16:45:43.363]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 109.59, "latitude": 37.32990718, "longitude": -122.02041136, "speed": 3.39}, "timestamp": 1587221143360.991}
[Sat Apr 18 2020 16:45:44.296]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 105.99, "latitude": 37.32989753, "longitude": -122.02037272, "speed": 3.39}, "timestamp": 1587221144293.8508}
[Sat Apr 18 2020 16:45:45.316]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 109.14, "latitude": 37.32988681, "longitude": -122.02033604, "speed": 3.38}, "timestamp": 1587221145314.0989}
[Sat Apr 18 2020 16:45:46.341]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 118.45, "latitude": 37.32987133, "longitude": -122.0202987, "speed": 3.59}, "timestamp": 1587221146338.483}
[Sat Apr 18 2020 16:45:47.341]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 111.1, "latitude": 37.32985921, "longitude": -122.02025905, "speed": 3.6}, "timestamp": 1587221147339.0642}
[Sat Apr 18 2020 16:45:48.260]  LOG      Geolocation new position: {"coords": {"accuracy": 10, "altitude": 0, "altitudeAccuracy": -1, "heading": 107.49, "latitude": 37.32984991, "longitude": -122.02021905, "speed": 3.61}, "timestamp": 1587221148257.746}
```

**Comment by Agontuk** [comment]
Ohh, I thought you're facing this in android. This library uses iOS module from [@react-native-community/react-native-geolocation](https://github.com/react-native-community/react-native-geolocation), so you should file an issue there instead.

**Comment by OlivierVoyer** [comment]
Added a comment in https://github.com/react-native-community/react-native-geolocation/issues/85#issuecomment-616061999

**Comment by Agontuk** [comment]
@OlivierVoyer, I'm trying to implement my own iOS module right now. I'll try to solve this issue in my implementation. Let me know if you're willing to test it, and your current xcode & RN version. You can track the progress here [https://github.com/Agontuk/react-native-geolocation-service/issues/173](https://github.com/Agontuk/react-native-geolocation-service/issues/173).

I'll update the setup instructions soon.

**Comment by Agontuk** [comment]
@OlivierVoyer , I've updated the instructions to setup the new ios module in this branch [next](https://github.com/Agontuk/react-native-geolocation-service/tree/next). If possible, please test and let me know if you have any trouble during setup. I've provided a possible fix for this issue, but right now background mode is not implemented yet. So you can only test it while the app is in foreground.

**Comment by OlivierVoyer** [comment]
Sorry for the delay, I tried to test the new branch but didn't manage to compile because of https://github.com/Agontuk/react-native-geolocation-service/issues/175#issuecomment-636502776

**Comment by Agontuk** [comment]
> Sorry for the delay, I tried to test the new branch but didn't manage to compile because of [#175 (comment)](https://github.com/Agontuk/react-native-geolocation-service/issues/175#issuecomment-636502776)

No need to test next branch, it's already published in v5.0.0. This issue should be fixed there, so closing this.

**Comment by OlivierVoyer** [comment]
v5.0.0 is fixing the main issue I described in this thread. Thanks @Agontuk
